ir/instruction.ts

```diff
@@ -0,0 +1,162 @@
+export interface LiteralValue {
```

Comment:
Is it important to use interfaces? It makes the isX() method a little bit brittle. If these where classes we could have a Symbol for each type, and a type variable with the correct symbol.

Reply:
I couldn't get it to work using Symbol, but I think this is the same idea.